src/EventTickets/resources/admin/components/TicketTypeFormModal/index.tsx

```diff
@@ -0,0 +1,100 @@
+import {SubmitHandler, useForm} from 'react-hook-form';
+import {__} from '@wordpress/i18n';
+import styles from './TicketTypeFormModal.module.scss';
+import FormModal from '../FormModal';
+import {useTicketTypeForm} from './ticketTypeFormContext';
+import {Inputs, TicketModalProps} from './types';
+import {useEffect} from 'react';
+import EventTicketsApi from '../api';
+
+/**
+ * Ticket Form Modal component
+ *
+ * @unreleased
+ */
+export default function TicketTypeFormModal({isOpen, handleClose, apiSettings, eventId}: TicketModalProps) {
+    const {ticketData} = useTicketTypeForm();
+    const API = new EventTicketsApi(apiSettings);
+
+    const {
+        register,
+        handleSubmit,
+        reset,
+        formState: {errors, isDirty},
+    } = useForm<Inputs>();
+
+    useEffect(() => {
+        reset({
+            title: ticketData?.title || '',
+            description: ticketData?.description || '',
+            price: ticketData?.price || null,
+            capacity: ticketData?.capacity || null,
+        });
+    }, [ticketData, reset]);
+
+    const onSubmit: SubmitHandler<Inputs> = async (data) => {
+        try {
+            data.price = data.price * 100;
```

Comment:
This assumes a non-zero decimal currency. Do we have a money library in JS that we can use to format this to a minor amount?

Reply:
```js
data.price = accounting.formatNumber(data.price, 2, '', '');
```
What do you think?

Reply:
I thought we were using something like [dinero.js](https://dinerojs.com/)

```js
const price = Dinero({ amount: 5000, currency: 'EUR' })
```

Reply:
How are we handling it in the Donation Form?

Reply:
Looks like we are using `Intl.NumberFormat` for the Donation Form.

Reply:
For backing into the minor amount from a locale formatted input we may need to check the currency for the configured precision and then move the decimal point (if necessary) from there.

Reply:
It seems that even using Dinero, we keep needing to manually multiply by 100 as it doesn't have a method to convert decimal number to amount.

```js
const priceInMinorAmount = Dinero({amount: Math.round(data.price * 100), currency: currencyCode}).getAmount();
data.price = priceInMinorAmount;
```

Reply:
What if instead we change the API endpoint to receive the price as a decimal then you handle Money amounts from there?

Reply:
That is an option - and may be the best option at this point.

Reply:
Wait. I think we are using a React Currency Input somewhere else that we could use here.

Reply:
I'm getting ahead of myself. For now let's take the easiest path forward and we can consider refactoring after QA.

Reply:
Well. Our input field currently only accept integers so I think that keep things as they are now is fine. Later, we'll replace that input with the React Currency Input Field component and many of this may change.

Reply:
That sounds good.